README.md

```diff
@@ -25,464 +24,423 @@ Features include:
+	// Declare a successful response that returns a slice of things
+	responses.OK().Headers("Foo").Model([]Thing{}),
+	// Errors automatically set the right status, content type, and model for you.
+	responses.InternalServerError(),
+).Run(func(ctx huma.Context) {
+	// This works because the `Foo` header was declared above.
+	ctx.Header().Set("Foo", "Some value")
+
+	// The `WriteModel` convenience method handles content negotiation and
+	// serializaing the response for you.
+	ctx.WriteModel(http.StatusOK, []Thing{
+		Thing{Name: "Test1"},
+		Thing{Name: "Test2"},
+	})
 
-```go
-func (id string) *MyModel { return &MyModel{ID: id} }
+	// Alternatively, you can write an error
+	ctx.WriteError(http.StatusInternalServerError, "Some message")
+})
 ```
 
-> :whale: Confused about what a handler should look like? Just run your service and it'll print out an approximate handler function when it panics.
+If you try to set a response status code or header that was not declared you will get a runtime error. If you try to call `WriteModel` or `WriteError` more than once then you will get an error because the writer is considered closed after those methods.
 
-## Parameters
+### Errors
 
-Huma supports three types of parameters:
+Errors use [RFC 7807](https://tools.ietf.org/html/rfc7807) and return a structure that looks like:
 
-- Required path parameters, e.g. `/things/{thingId}`
-- Optional query string parameters, e.g. `/things?q=filter`
-- Optional header parameters, e.g. `X-MyHeader: my-value`
+```json
+{
+  "status": 504,
+  "title": "Gateway Timeout",
+  "detail": "Problem with HTTP request",
+  "errors": [
+    {
+      "message": "Get \"https://httpstat.us/418?sleep=5000\": context deadline exceeded"
+    }
+  ]
+}
+```
 
-Optional parameters require a default value.
+The `errors` field is optional and may contain more details about which specific errors occurred.
 
-Here is an example of an `id` parameter:
+It is recommended to return exhaustive errors whenever possible to prevent user frustration with having to keep retrying a bad request and getting back a different error. The context has `AddError` and `HasError()` functions for this:
 
 ```go
-r.Resource("/notes",
-	huma.PathParam("id", "Note ID"),
-	huma.ResponseError(404, "Note was not found"),
-	huma.ResponseJSON(200, "Success"),
-).
-Get("Get a note by its ID", func(id string) (*huma.ErrorModel, *Note) {
-	// Implementation goes here
+app.Resource("/exhaustive").Get("exhaustive", "Exhastive errors example",
+	responses.OK(),
+	responses.BadRequest(),
+).Run(func(ctx huma.Context) {
+	for i := 0; i < 5; i++ {
+		// Use AddError to add multiple error details to the response.
+		ctx.AddError(fmt.Errorf("Error %d", i))
+	}
+
+	// Check if the context has had any errors added yet.
+	if ctx.HasError() {
+		// Use WriteError to set the actual status code, top-level message, and
+		// any additional errors. This sends the response.
+		ctx.WriteError(http.StatusBadRequest, "Bad input")
```

Comment:
> 1. There is no explicit return OK-type of statement, which I'd expect. Do handlers default to returning OK?

Handlers don't return anything now (e.g. to support streaming), so there is no explicit `return` statement anymore. Like the standard Go library, the default response is `http.StatusOK` with an empty body if you never write anything. If you call on any of the `Write*` methods that will explicitly set the response info, e.g. `ctx.WriteHeader(http.StatusOK)`.

> 2. If I declare that a valid response is a response.RedirectPermenantly(), it isn't clear to me from this example how I'd know which http.Status* field I should use.

They all deliberately match, so `responses\.(.*)` ⇨ `http.Status$1`, e.g. `http.StatusRedirectPermanently`. The `responses` package is just a convenience layer to make response declaration simpler. For example:

```go
// This:
responses.RedirectPermanently()

// Is equal to:
huma.NewResponse(http.StatusRedirectPermanently, http.StatusText(http.StatusRedirectPermanently))
```

Error responses also add a content type and model for you. Overall the goal is to improve readability and easy-of-use, but you can fall back to `huma.NewResponse` for more complex scenarios if needed.

> Since ctx includes a response writer, I'd expect to be able to do something like:
>
> ```
> ctx.response.BadRequest("BadInput")
> ```
> Which looks more similar to the declared response

It's specifically an [`http.ResponseWriter`](https://golang.org/pkg/net/http/#ResponseWriter) so it only supports `Header()`, `WriteHeader`, and `Write` calls just like the standard library. We only add the `WriteModel` and `WriteError` calls for convenience. We could add more convenience and increase the size of the interface, but in general you want to keep the interfaces small. I'm curious to hear if others find this confusing.

Reply:
> They all deliberately match, so responses\.(.*) ⇨ http.Status$1, e.g. http.StatusRedirectPermanently. 

Something to this effect should be added to the readme